pkg/apis/picchu/v1alpha1/incarnation_types.go

```diff
@@ -27,12 +60,33 @@ type IncarnationList struct {
 	Items           []Incarnation `json:"items"`
 }
 
+func (i *IncarnationList) LatestRelease() *Incarnation {
+	releases := i.SortedReleases()
+	if len(releases) > 0 {
+		return &releases[0]
+	}
+	return nil
+}
+
+func (i *IncarnationList) SortedReleases() []Incarnation {
+	releases := []Incarnation{}
+	for _, incarnation := range i.Items {
+		if incarnation.Spec.Release.Eligible {
+			releases = append(releases, incarnation)
+		}
+	}
+	sort.Slice(releases, func(i, j int) bool {
+		return releases[i].GitTimestamp().After(releases[j].GitTimestamp())
```

Comment:
we should raise some kind of alarm if `GitTimestamp().IsZero()`, or a bug we introduce later will cause revisions to be released in an arbitrary order